src/LCRmeter-Keithley_4200-SCS/main.py

```diff
@@ -0,0 +1,348 @@
+                tcp_ip_port = self.port_string[11:].strip()  # removing "LPTlib via "
+                tcp_ip_port_splitted = tcp_ip_port.split(":")  # in case a port is given
+                tcp_ip = tcp_ip_port_splitted[0]
+
+                tcp_port = int(tcp_ip_port_splitted[1]) if len(tcp_ip_port_splitted) == 2 else 8888
+
+                self.lpt = ProxyClass.Proxy(tcp_ip, tcp_port, "lpt")
+                # not supported yet with pylptlib -> # self.inst = Proxy(tcp_ip, tcp_port, "inst")
+                self.param = ProxyClass.Proxy(tcp_ip, tcp_port, "param")
+
+            else:
+                # we directly use pylptlib
+                self.lpt = lpt
+                # not supported yet with pylptlib -> # self.inst = inst
+                self.param = param
+
+            self.lpt.initialize()
+
+            self.card_id = self.lpt.getinstid(self.card_name)
+
+    def initialize(self) -> None:
+        """Initialize the Keithley 4200-SCS LCRmeter."""
+        # check for correct use of sweep mode and step mode
+        if self.stepmode != "None" and self.sweepmode == self.stepmode:
+            msg = "Sweep and step mode cannot be the same."
+            raise Exception(msg)
+
+        self.identifier = "Keithley_4200-SCS_" + self.port_string
+
+        if self.identifier not in self.device_communication:
+            if self.command_set == "LPTlib":
+                self.lpt.tstsel(1)
+                self.lpt.devint()  # This command resets all active instruments in the system to their default states.
+
+            elif self.command_set == "US":
+                pass  # not used at the moment because KXCI communication is not supported yet
+                # if self.current_range != "Auto":
+                #     msg = "When using KXCI only Auto current range is supported."
+                #     raise Exception(msg)
+                #
+                # self.clear_buffer()
+                # self.set_to_4200()
+                # self.set_command_mode("US")
+                # self.set_data_service()
+                # self.set_resolution(7)
+
+            self.device_communication[self.identifier] = {}  # dictionary that can be filled with further information
+
+    def deinitialize(self) -> None:
+        """Reset device and close connection."""
+        if self.identifier in self.device_communication:
+            if self.command_set == "LPTlib":
+                self.lpt.devint()  # restores default values
+                self.lpt.tstdsl()  # deselects test station
+
+            del self.device_communication[self.identifier]
+
+    def configure(self) -> None:
+        """Set bias and measurement parameters with start values from GUI."""
+        self.set_frequency(self.frequency)
+        self.set_delay(self.trigger_delay)
+
+        self.set_dc_bias(self.value_bias)
+        self.set_ac_voltage(self.value_level_rms)
+
+        self.set_measure_range()
```

Comment:
set_measure_range uses the self.range being "Auto". This handover of information via class variables is often a bit hidden and not easy to read. 
An alternative would be self.set_measure_range(self.range) and the function expects a string like "Auto" or enum like 0 (or both can be processed).